mlir/lib/Dialect/AMDGPU/IR/AMDGPUDialect.cpp

```diff
@@ -506,6 +506,40 @@ LogicalResult GatherToLDSOp::verify() {
   return success();
 }
 
+LogicalResult ScaledMFMAOp::verify() {
+  unsigned opselA = getOpselA() >> 8;
+  unsigned opselB = getOpselB() >> 8;
+
+  if (opselA != 0)
+    return emitOpError("Opsel A must be a zero extended 8 bit value");
```

Comment:
No

`opselA` and `opselB` are `0 <= i < 4` ... and I think there's a way to specify this constraint in Tablegen